pkg/osbuild/oscap_autotailor_stage.go

```diff
@@ -11,29 +11,57 @@ type OscapAutotailorStageOptions struct {
 	Config   OscapAutotailorConfig `json:"config"`
 }
 
-type OscapAutotailorConfig struct {
-	TailoredProfileID string   `json:"new_profile"`
-	Datastream        string   `json:"datastream"`
-	ProfileID         string   `json:"profile_id"`
-	Selected          []string `json:"selected,omitempty"`
-	Unselected        []string `json:"unselected,omitempty"`
+type OscapAutotailorConfig interface {
+	validate() error
+	isAutotailorConfig()
 }
 
-func (OscapAutotailorStageOptions) isStageOptions() {}
+type AutotailorKeyValueConfig struct {
+	NewProfile string   `json:"new_profile,omitempty"`
+	Datastream string   `json:"datastream"`
+	ProfileID  string   `json:"profile_id"`
+	Selected   []string `json:"selected,omitempty"`
+	Unselected []string `json:"unselected,omitempty"`
+}
+
+func (c AutotailorKeyValueConfig) isAutotailorConfig() {}
 
-func (c OscapAutotailorConfig) validate() error {
+func (c AutotailorKeyValueConfig) validate() error {
 	if c.Datastream == "" {
 		return fmt.Errorf("'datastream' must be specified")
 	}
+	if c.NewProfile == "" {
+		return fmt.Errorf("'new_profile' must be specified")
+	}
```

Comment:
The stage schema does not require it to be specified AFAICT and the struct definitions also marks it as `omitempty`... :thinking:

Reply:
> stage schema does not require it

I think this is a mistake and I'll need to fix it in `osbuild`. It needs to be one of `new_profile` or `tailored_profile_id`

Reply:
Here's a PR to fix that: https://github.com/osbuild/osbuild/pull/1823

Reply:
You should update the osbuild ref in `Schutzfile` once the PR is merged, just to be sure.